include/fg/chart.h

```diff
@@ -0,0 +1,191 @@
+/*******************************************************
+ * Copyright (c) 2015-2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <fg/defines.h>
+#include <fg/image.h>
+#include <fg/plot.h>
+#include <fg/surface.h>
+#include <fg/histogram.h>
+
+#include <string>
+#include <vector>
+
+namespace internal
+{
+class _Chart;
+}
+
+namespace fg
+{
+
+/**
+   \class Chart
+
+   \brief Chart is base canvas where other plottable objects are rendered.
+
+   Charts come in two types:
+        - \ref FG_2D - Two dimensional charts
+        - \ref FG_3D - Three dimensional charts
+ */
+class Chart {
+    private:
+        ChartType mChartType;
+        internal::_Chart* mValue;
+
+    public:
+        /**
+           Creates a Chart object with given dimensional property
+
+           \param[in] cType is chart dimension property
+         */
+        FGAPI Chart(const ChartType cType);
+
+        /**
+           Chart destructor
+         */
+        FGAPI ~Chart();
+
+        /**
+           Set axes titles for the chart
+
+           \param[in] pX is x-axis title label
+           \param[in] pY is y-axis title label
+           \param[in] pZ is z-axis title label
+         */
+        FGAPI void setAxesTitles(const std::string pX,
```

Comment:
@umar456 is this safe ? I remember you mentioning something about C++ classes changing between implementations. Would this cause an issue ?